### PR TITLE
feat(isort): Implement isort.force_to_top

### PR DESCRIPTION
**Pull request by spaceone**
test what `isort` does via:
`isort -t "lib1" -t "lib3" -t "lib5" -t "lib3.lib4" -t "z" crates/ruff/resources/test/fixtures/isort/force_to_top.py`

Issue #2695

### CODE REVIEW

File: crates/ruff/src/rules/isort/sorting.rs [hunk 1]
```diff
@@ -100,11 +114,22 @@ pub fn cmp_import_from(
...
+        //).then_with(||
+        //match (
+        //    force_to_top.contains(&import_from1.module.unwrap()),
+        //    force_to_top.contains(&import_from2.module.unwrap()),
+        //) {
+        //    (true, true) => Ordering::Equal,
+        //    (false, false) => Ordering::Equal,
+        //    (true, false) => Ordering::Greater,
+        //    (false, true) => Ordering::Less,
+        //}
```
Comment (charliermarsh) [inline crates/ruff/src/rules/isort/sorting.rs:132]: Are the commented-out lines included by accident?

Comment (spaceone) [reply crates/ruff/src/rules/isort/sorting.rs:132]: no, on purpose currently. I am not sure if we should sort `from`-imports also there. From reading the `isort` code it just splits anything at ` ` and looks if this is in `force_to_top`.
Also the commented code had one error which I don't remember where I couldn't find a rust-solution.

Comment (spaceone) [reply crates/ruff/src/rules/isort/sorting.rs:132]: `isort` also doesn't really have much tests for `force-to-top`, so I have to reverse engineer a little how it handels `from`-imports.

Comment (spaceone) [reply crates/ruff/src/rules/isort/sorting.rs:132]: I think the problem was that I don't know how to convert a `import_from1.module` into a string. maybe you can help here?

Comment (charliermarsh) [reply crates/ruff/src/rules/isort/sorting.rs]: Does this trait help at all?

```rust
pub trait Importable {
    fn module_name(&self) -> String;
    fn module_base(&self) -> String;
}

impl Importable for AliasData<'_> {
    fn module_name(&self) -> String {
        self.name.to_string()
    }

    fn module_base(&self) -> String {
        self.module_name().split('.').next().unwrap().to_string()
    }
}

impl Importable for ImportFromData<'_> {
    fn module_name(&self) -> String {
        ast::helpers::format_import_from(self.level, self.module)
    }

    fn module_base(&self) -> String {
        self.module_name().split('.').next().unwrap().to_string()
    }
}
```

**Pull request merged**